devdoc/interlocked_requirements.md

```diff
@@ -0,0 +1,298 @@
+# interlocked
+================
+
+## Overview
+
+`interlocked` exports platform independent atomic operations. It is a platform abstraction and it requires a specific implementation for each platform.
+
+The interface is based on the Windows Interlocked API.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_add, volatile int32_t*, addend, int32_t, value);
+MOCKABLE_FUNCTION(, int32_t, interlocked_and, volatile int32_t*, destination, int32_t, value);
+MOCKABLE_FUNCTION(, int16_t, interlocked_and_16, volatile int16_t*, destination, int16_t, value);
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_64, volatile int64_t*, destination, int64_t, value);
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_8, volatile int8_t*, destination, int8_t, value);
+MOCKABLE_FUNCTION(, int32_t, interlocked_compare_exchange, volatile int32_t*, destination, int32_t, exchange, int32_t, comperand);
+MOCKABLE_FUNCTION(, bool, interlocked_compare_exchange_128, volatile int64_t*, destination, int64_t, exchange_low, int64_t, exchange_high, int64_t, comperand_result);
```

Comment:
>int64_t [](start = 139, length = 7)

[https://docs.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedcompareexchange128](https://docs.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedcompareexchange128) seems to point that comperand_result is a pointer actually.

```code
BOOLEAN InterlockedCompareExchange128(
  LONG64 volatile *Destination,
  LONG64          ExchangeHigh,
  LONG64          ExchangeLow,
  LONG64          *ComparandResult
);
```

So i guess this needs to be updated